client/src/controllers/DropdownController.test.js

```diff
@@ -0,0 +1,41 @@
+import { Application } from '@hotwired/stimulus';
+import { DropdownController } from './DropdownController';
+
+describe('DropdownController', () => {
+  let application;
+
+  beforeAll(() => {
+    application?.stop();
+
+    document.body.innerHTML = `
+<div data-controller="w-dropdown">
+  <button type="button" data-w-dropdown-target="toggle" aria-label="Actions"></button>
+  <div data-w-dropdown-target="content">
+    <a href="/">Option</a>
+  </div>
+</div>`;
+
+    application = Application.start();
+    application.register('w-dropdown', DropdownController);
+  });
+
+  it('initialises Tippy.js on connect', () => {
+    const toggle = document.querySelector('[data-w-dropdown-target="toggle"]');
+    const content = document.querySelector(
+      '[data-w-dropdown-target="content"]',
+    );
+    expect(toggle.getAttribute('aria-expanded')).toBe('false');
+    expect(content).toBe(null);
+  });
+
+  it('triggers custom event on activation', () => {
+    const toggle = document.querySelector('[data-w-dropdown-target="toggle"]');
+    const mock = jest.fn();
+    document.addEventListener('w-dropdown:shown', mock);
+    toggle.dispatchEvent(new Event('click'));
+    // Leave time for animation to complete.
+    setTimeout(() => {
```

Comment:
Just a heads up, this unit test will never run, adding something like `expect(true).toBe(false)` would never fail the test.

I have a PR with a fix here https://github.com/wagtail/wagtail/pull/10617

Turns out to be a bit more nuanced, normally setTimeout in a test will not really do anything, but using `jest.useFakeTimers()` did not solve the problem here. Tippy relies on `transitionend` events, which will not get fired in JSDom.